_posts/2019-09-23-exposing-http-bridge.md

```diff
@@ -0,0 +1,270 @@
+What you need in this case is creating an `Ingress` resource defining the rule for routing the HTTP traffic to the Strimzi Kafka Bridge.
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: my-bridge-ingress
+spec:
+  rules:
+  - host: my-bridge.io
+    http:
+      paths:
+      - path: /
+        backend:
+          serviceName: my-bridge-bridge-service
+          servicePort: 8080
+```
+
+When the above resource is created, the Strimzi Kafka Bridge is reachable through the `my-bridge.io` host so you can interact with it using different HTTP methods at the address `http://my-bridge.io:80/<endpoint>` where `<endpoint>` is one of the REST endpoints exposed by the bridge for sending and receiving messages, subscribing to topics and so on.
+
+> If your cluster is running using Minikube, don't forget to update your local `/etc/hosts` file adding a line as `<minikubeip> my-bridge.io` where you can get the `<minikubeip>` by running the command `minikube ip` first.
+
+In order to verify that the Ingress is working properly, try to hit the `/healthy` endpoint of the bridge with the following `curl` command.
+
+```shell
+curl -v GET http://my-bridge.io/healthy
+```
+
+If the bridge is reachable through the Ingress, it will return an HTTP response with status code `200 OK` but an empty body.
+
+# Producing messages
+
+Assuming that the Kafka brokers have topic auto creation enabled, we can start immediately to send messages through the `/topic/{topicname}` endpoint exposed by the HTTP bridge.
+
+The HTTP request payload is always a JSON but the message values can be JSON or binary (encoded in base64 because you are sending binary data in a JSON payload so encoding in a string format is needed).
+
+```shell
+curl -X POST \
+  http://my-bridge.io/topics/my-topic \
+  -H 'content-type: application/vnd.kafka.json.v2+json' \
+  -d '{
+    "records": [
+        {
+            "key": "key-1",
+            "value": "value-1"
+        },
+        {
+            "key": "key-2",
+            "value": "value-2"
+        }
+    ]
+}'
+```
+
+After writing the messages into the topic, the bridge replies with an HTTP status code `200 OK` and a JSON paylod describing in which partition and at which offset the messages are written.
+In this case, the auto created topic has just one partition and the reply can look like the following one.
+
+```json
+{ 
+   "offsets":[ 
+      { 
+         "partition":0,
+         "offset":0
+      },
+      { 
+         "partition":0,
+         "offset":1
+      }
+   ]
+}
+```
+
+# Consuming messages
+
+Consuming messages is not so simple as producing because there are several steps to do which involve different endpoints.
+First of all, creating a consumer through the `/consumers/{groupid}` endpoint doing an HTTP POST with a body containing some of the supported configuration parameters, the name of the consumer and the data format (JSON or binary).
```

Comment:
I wonder if this deserves a bit more explanation. You are creating a consumer, but you say the address is `/consumers/{groupid}` with the group ID in it and the consumer name in the payload. I don't think this is really REST native. I'm always confused by this myself - i wonder if this deserves more explanation - if there is any.